File: app/src/main/java/ru/fylmr/diploma/crypto/primitives/curve25519/ge_p3_to_p2.java
package ru.fylmr.diploma.crypto.primitives.curve25519;

public class ge_p3_to_p2 {

//CONVERT #include "ge.h"

/*
r = p
*/

    public static void ge_p3_to_p2(ge_p2 r, GroupElemExtended p) {
        new fe_copy(r.X, p.X);
        new fe_copy(r.Y, p.Y);
        new fe_copy(r.Z, p.Z);
    }


}

File: app/src/main/java/ru/fylmr/diploma/crypto/primitives/curve25519/ge_p3_to_p2.kt
package ru.fylmr.diploma.crypto.primitives.curve25519

/**
 * ge_p3_to_p2
 */
fun ge_p3_to_p2(r: ge_p2, p: GEExtended) {
    fe_copy(r.X, p.X)
    fe_copy(r.Y, p.Y)
    fe_copy(r.Z, p.Z)
}
